internal/entities/command_search.go

```diff
@@ -12,61 +12,44 @@ import (
 	"github.com/newrelic/newrelic-client-go/v2/pkg/entities"
 )
 
+var (
+	entitySearchCaseSensitive bool
+)
+
 var cmdEntitySearch = &cobra.Command{
 	Use:   "search",
 	Short: "Search for New Relic entities",
 	Long: `Search for New Relic entities
 
 The search command performs a search for New Relic entities.
 `,
-	Example: "newrelic entity search --name <applicationName>",
+	Example: "newrelic entity search --name=<name> --type=<type> --domain=<domain> --tags=tagKey1:tagValue2,tagKey2:tagValue2",
 	PreRun:  client.RequireClient,
 	Run: func(cmd *cobra.Command, args []string) {
-		params := entities.EntitySearchQueryBuilder{}
-
-		if entityName == "" && entityType == "" && entityAlertSeverity == "" && entityDomain == "" {
+		if entityName == "" && entityType == "" && entityAlertSeverity == "" && entityDomain == "" && len(entityTags) == 0 {
 			utils.LogIfError(cmd.Help())
-			log.Fatal("one of --name, --type, --alert-severity, or --domain are required")
-		}
-
-		if entityName != "" {
-			params.Name = entityName
-		}
-
-		if entityType != "" {
-			params.Type = entities.EntitySearchQueryBuilderType(entityType)
+			log.Fatal("one of --name, --type, --alert-severity, --domain, or --tags is required")
 		}
 
-		if entityAlertSeverity != "" {
-			params.AlertSeverity = entities.EntityAlertSeverity(entityAlertSeverity)
-		}
+		tags, err := entities.ConvertTagsToMap(entityTags)
+		utils.LogIfError(err)
 
-		if entityDomain != "" {
-			params.Domain = entities.EntitySearchQueryBuilderDomain(entityDomain)
+		searchParams := entities.EntitySearchParams{
+			Name:            entityName,
+			Domain:          entityDomain,
+			Type:            entityType,
+			AlertSeverity:   entityAlertSeverity,
+			Tags:            tags,
+			IsCaseSensitive: entitySearchCaseSensitive,
 		}
 
-		if entityTag != "" {
-			key, value, err := assembleTagValue(entityTag)
-			utils.LogIfFatal(err)
-
-			params.Tags = []entities.EntitySearchQueryBuilderTag{{Key: key, Value: value}}
-		}
-
-		if entityReporting != "" {
-			reporting, err := strconv.ParseBool(entityReporting)
-
-			if err != nil {
-				log.Fatalf("invalid value provided for flag --reporting. Must be true or false.")
-			}
-
-			params.Reporting = reporting
-		}
+		query := entities.BuildEntitySearchNrqlQuery(searchParams)
 
-		results, err := client.NRClient.Entities.GetEntitySearchWithContext(
-			utils.SignalCtx,
-			entities.EntitySearchOptions{},
-			"",
-			params,
+		results, err := client.NRClient.Entities.GetEntitySearchByQueryWithContext(context.Background(),
+			entities.EntitySearchOptions{
+				CaseSensitiveTagMatching: entitySearchCaseSensitive,
```

Comment:
A thought I'd like to share (not for now, maybe sometime in future for one of us to add) - `entities.EntitySearchOptions` also has a `limit` field that helps specify the number of entities to be returned in results - do you think it would be a good idea for us to facilitate this via a new flag in the `newrelic entity search` command